utils/pretty.py

```diff
@@ -24,13 +28,60 @@ def x(o):
     return x
 
 
+def process_attrs(attrs):
```

Comment:
Have the interface for *_attrs be clear, which is that they're lists, then there's no ambiguity or uncertainty.